src/test/ui/borrowck/anonymous-region-in-apit.stderr

```diff
@@ -0,0 +1,16 @@
+error[E0521]: borrowed data escapes outside of closure
+  --> $DIR/anonymous-region-in-apit.rs:8:17
+   |
+LL | fn qux(foo: impl Foo<&str>) {
```

Comment:
Is there any easy way to get the `def_id` of the lifetime itself and point to it?
r=me if not.

Reply:
No, as far as I understand it, the way that borrowck does it requires walking through HIR, and I didn't want to reinvent that for just this case.